### DISCUSSION TITLE
PV energy as sensor?

### DISCUSSION THREAD

**Discussion by Reudo** [marq24/ha-evcc#36] [Ideas]
Hello, I don't know if evcc already gets the PV energy of my inverter so maybe the question / idea is better suited for the main evcc github.

At the moment I use an integral sensor with the sensor.evcc_pv_power to get my energy but I've noticed that can be a little inaccurate.
I am using a SMA Inverter with speedwire and evcc won't work if I try to use the sma integration simultaneously.

So it would be nice to just get the PV Energy sensor directly.

**Comment by marq24** [comment]
when you have HA installation, then IMHO it's the best approach to let evcc fetch the data from your HA sensors (instead of letting evcc fetch the data from the hardware)... -> https://github.com/marq24/ha-evcc/blob/main/HA_AS_EVCC_SOURCE.md

**Reply by Reudo** [reply]
I see. Didn't know that is possible. Maybe I'll do that then. Ty
